lib/streamlit/runtime/scriptrunner/script_runner.py

```diff
@@ -446,24 +445,33 @@ def _run_script(self, rerun_data: RerunData) -> None:
 
             fragment_ids_this_run = set(rerun_data.fragment_id_queue)
 
+            ctx = self._get_script_run_ctx()
             # Clear widget state on page change. This normally happens implicitly
             # in the script run cleanup steps, but doing it explicitly ensures
             # it happens even if a script run was interrupted.
             try:
-                old_hash = self._session_state[SCRIPT_RUN_PAGE_SCRIPT_HASH_KEY]
+                old_hash = ctx.page_script_hash
             except KeyError:
                 old_hash = None
```

Comment:
Is the `try-except` still relevant if we access the `ctx` dataclass instead of a the session state dict?
If yes, I would suggest to update the `page_script_hash` property function [here](https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/runtime/scriptrunner/script_run_context.py#L92) to do the check and return `None` if the property is not set / does not exist. If you modify it, could you also please add type annotations to the [page_script_hash property function](https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/runtime/scriptrunner/script_run_context.py#L92) 😇?

Reply:
Good point. I can remove the try-except